contracts/libraries/VaultLifecycleSTETH.sol

```diff
@@ -353,66 +353,74 @@ library VaultLifecycleSTETH {
+        }
 
-        uint256 wstETHPerStETH = IWSTETH(collateralToken).tokensPerStEth();
-        uint256 amountToUnwrap = stethNeeded.mul(wstETHPerStETH).div(10**18);
+        // Scenario 2
+        // Now that we satisfied the ETH + stETH sum, we swap the stETH amounts necessary
+        // to facilitate a withdrawal
 
-        if (amountToUnwrap > 0) {
-            IWSTETH wsteth = IWSTETH(collateralToken);
-            IERC20 steth = IERC20(stethToken);
+        // This won't underflow since we already asserted that ethAvailable < amount before this
+        uint256 stEthAmountToSwap =
+            DSMath.min(amount.sub(ethAvailable), stethAvailable);
 
-            uint256 startStethBalance = steth.balanceOf(address(this));
+        steth.safeApprove(crvPool, stEthAmountToSwap);
 
-            if (stethNeeded > startStethBalance) {
-                amountToUnwrap = stethNeeded
-                    .sub(startStethBalance)
-                    .mul(wstETHPerStETH)
-                    .div(10**18);
-                // Unwrap to stETH
-                wsteth.unwrap(amountToUnwrap);
-            }
+        // CRV SWAP HERE from steth -> eth
+        // 0 = ETH, 1 = STETH
+        // We are setting 1, which is the smallest possible value for the _minAmountOut parameter
+        // However it is fine because we check that the totalETHOut >= minETHOut at the end
+        // which makes sandwich attacks not possible
+        uint256 ethAmountOutFromSwap =
+            ICRV(crvPool).exchange(1, 0, stEthAmountToSwap, 1);
 
-            // Post-unwrap, the stETH balance will not completely match the stethNeeded
-            // due to precision issues.
-            // E.g. 0.5 ETH is 499999999999999998 instead of 500000000000000000
-            // We just send the entire stETH balance for the swap
-            uint256 stETHAmount =
-                steth.balanceOf(address(this)).sub(startStethBalance);
-
-            // approve steth exchange
-            steth.safeApprove(crvPool, stETHAmount);
-
-            // CRV SWAP HERE from steth -> eth
-            // 0 = ETH, 1 = STETH
-            // We are setting 1, which is the smallest possible value for the _minAmountOut parameter
-            // However it is fine because we check that the amountETHOut >= minETHOut at the end
-            // which makes sandwich attacks not possible
-            uint256 swappedAmount =
-                ICRV(crvPool).exchange(1, 0, stETHAmount, 1);
-
-            amountETHOut = amountETHOut.add(swappedAmount);
-        }
+        uint256 totalETHOut = ethAvailable.add(ethAmountOutFromSwap);
 
-        // This revert does not account for the ETH that is already unwrapped
         // Since minETHOut is derived from calling the Curve pool's getter,
         // it reverts in the worst case where the user needs to unwrap and sell
         // 100% of their ETH withdrawal amount
         require(
-            amountETHOut >= minETHOut,
+            totalETHOut >= minETHOut,
```

Comment:
If `minETHOut` out is set to 0 by the caller of this function, a MEV bot will hijack all the `stEth` during the curve swap.

Reply:
Addressed by asserting that minETHOut has to be at least 0.95 * amount